src/validator-integration.js

```diff
@@ -34,14 +35,24 @@ export function maybeValidate(win) {
   }
 
   /** @const {!Element} */
-  const s = win.document.createElement('script');
+  const validatorScript = win.document.createElement('script');
   // TODO(@cramforce): Introduce a switch to locally built version for local
   // development.
-  s.src = `${urls.cdn}/v0/validator.js`;
-  s.onload = () => {
-    win.document.head.removeChild(s);
-    /* global amp: false */
-    amp.validator.validateUrlAndLog(filename, win.document, getMode().filter);
-  };
-  win.document.head.appendChild(s);
+  validatorScript.src = `${urls.cdn}/v0/validator.js`;
+
+  const examinerScript = win.document.createElement('script');
+  examinerScript.src = `${urls.cdn}/examiner0.js`;
+
+  Promise.all([
```

Comment:
the examiner.js script might fail the validator. we need to remove it from DOM before run validate()